src/govuk/components/exit-this-page/exit-this-page.mjs

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable es-x/no-function-prototype-bind -- Polyfill imported */
+
+import { normaliseDataset } from '../../common/normalise-dataset.mjs'
+import { nodeListForEach, mergeConfigs, extractConfigByNamespace } from '../../common.mjs'
+import { I18n } from '../../i18n.mjs'
+import '../../vendor/polyfills/Element/prototype/classList.mjs'
+import '../../vendor/polyfills/Element/prototype/dataset.mjs'
+import '../../vendor/polyfills/Function/prototype/bind.mjs'
+
+/**
+ * @constant
+ * @type {ExitThisPageTranslations}
+ * @see Default value for {@link ExitThisPageConfig.i18n}
+ * @default
+ */
+var EXIT_THIS_PAGE_TRANSLATIONS = {
+  activated: 'Loading.',
+  timedOut: 'Exit this page expired.',
+  pressTwoMoreTimes: 'Shift, press 2 more times to exit.',
+  pressOneMoreTime: 'Shift, press 1 more time to exit.'
+}
+
+/**
+ * JavaScript functionality for the Exit this Page component
+ *
+ * @class
+ * @param {HTMLElement} $module - HTML element that wraps the EtP button
+ * @param {ExitThisPageConfig} [config] - Exit this Page config
+ */
+function ExitThisPage ($module, config) {
+  this.$module = $module
+
+  var defaultConfig = {
+    i18n: EXIT_THIS_PAGE_TRANSLATIONS
+  }
+
+  this.config = mergeConfigs(
```

Comment:
Mind adding all the JSDoc types like we do in the other components?

You'll start to see all the JavaScript type linting come to life and reveal problems if they exist

```suggestion  
  /**
   * @private
   * @type {ExitThisPageConfig}
   */
  this.config = mergeConfigs(
```